### DISCUSSION TITLE
Feedback on new Website

### DISCUSSION THREAD

**Discussion by phil-opp** [dora-rs/dora#298] [General]
Thanks to @haixuanTao, the dora project has a new website at https://dora.carsmos.ai/. Let's collect feedback here!

**Comment by phil-opp** [comment]
Some random things I noticed:

- There are lots of buttons with the text _"Learn More"_ on the front page. In my opinion, it would be better to give them a more descriptive text.
    - For example, the top-level button could say e.g. "Read our Guides" or "Get Started" instead.
    - Perhaps we could also add buttons to go directly to the Python and Rust API definitions?
    - The buttons for the different features could have the title of the target page as text. For example, the button under "Fast Prototyping" could have the title "Hot Reloading".
- Guides
    - Installation
        - We should link to the GitHub releases page to allow manual download+unzip (instead of using a shell script)
        - `cargo install` automatically reinstalls when there is a newer version. We probably don't want to use ‒force to force a reinstall even if the version is up-to-date.
    - Getting Started
        - Step 1: We should explcitly mention that one also needs to add dora to the PATH and open a new shell (I suspect that this will be a common source of errors).
        - Clarify that dataflow ID (and name) is randomly generated
        - Webcam plot: Mention that it's also possible to download the python files and specify a path instead of specifying the them by URL.
        - Yolov5: Add explanation what we see in the images (e.g. bounding box around detected persion)
- Reference
    - MLA is not implemented yet, so I would not include it in our reference at this point.
    - The "Framework" document is looks like an outdated design proposal, so I wouldn't include it either.
    - The design overview needs to be updated to include the dora daemon.
    - State management is not implemented yet, so including it in the reference already seems confusing.

I also opened https://github.com/dora-rs/dora-rs.github.io/pull/1 with some minor improvements.

**Reply by haixuanTao** [reply]
> There are lots of buttons with the text "Learn More" on the front page. In my opinion, it would be better to give them a more descriptive text.
> 
>  For example, the top-level button could say e.g. "Read our Guides" or "Get Started" instead.
>  The buttons for the different features could have the title of the target page as text. For example, the button under "Fast Prototyping" could have the title "Hot Reloading".

Absolutely, I'll make the change.

>  Perhaps we could also add buttons to go directly to the Python and Rust API definitions?

in which part of the frontpage do you want to add API buttons: benchmark, features, or within a new section?

> Guides
> 
>   Installation
>  We should link to the GitHub releases page to allow manual download+unzip (instead of using a shell script)

Absolutely, I'll make the change.

>  cargo install automatically reinstalls when there is a newer version. We probably don't want to use ‒force to force a reinstall even if the version is up-to-date.

True, thanks!

> Getting Started
> 
>     Step 1: We should explcitly mention that one also needs to add dora to the PATH and open a new shell (I suspect that this will be a common source of errors).

I feel like this is more of a good installation requirement. I can add a comment in the installation section.

>     Clarify that dataflow ID (and name) is randomly generated

ok!

>     Webcam plot: Mention that it's also possible to download the python files and specify a path instead of specifying the them by URL.

I feel like this is kind of a more advance feature. Also, there is quite a bit of internet issue in China, making Github Downloading kind of a pain point. What about a new page in the guide about web hosting?

>     Yolov5: Add explanation what we see in the images (e.g. bounding box around detected persion)

Ok, make sense!

**Reply by phil-opp** [reply]
> > Perhaps we could also add buttons to go directly to the Python and Rust API definitions?
> 
> in which part of the frontpage do you want to add API buttons: benchmark, features, or within a new section?

My idea was to add them next to the other button at the top level (as people might need it often). For example, like this:

![image](https://github.com/dora-rs/dora/assets/1131315/88fc03ae-c265-4ffc-afa7-df3a5de725a0)




> > Getting Started
> > ```
> > Step 1: We should explcitly mention that one also needs to add dora to the PATH and open a new shell (I suspect that this will be a common source of errors).
> > ```
> 
> I feel like this is more of a good installation requirement. I can add a comment in the installation section.

True. Perhaps we should refer to the "Installation" page in "Step 1"?



> > ```
> > Webcam plot: Mention that it's also possible to download the python files and specify a path instead of specifying the them by URL.
> > ```
> 
> 
> I feel like this is kind of a more advance feature. Also, there is quite a bit of internet issue in China, making Github Downloading kind of a pain point. What about a new page in the guide about web hosting?

What I meant is that the example at https://dora.carsmos.ai/docs/guides/getting-started/webcam_plot currently uses URL sources, which might be confusing to people because the other examples only use normal paths as sources.

**Reply by haixuanTao** [reply]
> What I meant is that the example at https://dora.carsmos.ai/docs/guides/getting-started/webcam_plot currently uses URL sources, which might be confusing to people because the other examples only use normal paths as sources.

My bad sorry!

**Comment by haixuanTao** [comment]
Add another end-2-end guide to autonomous driving using dora-drives

**Reply by haixuanTao** [reply]
Added to the website: https://dora.carsmos.ai/docs/guides/autonomous-driving/

**Comment by haixuanTao** [comment]
Add more disclaimer about the experimental aspect of dora

**Reply by haixuanTao** [reply]
Added disclaimer in https://github.com/dora-rs/dora/pull/316/commits/3b0f69f07de8ce434090a9ed1627dc030e99c3ee

**Comment by haixuanTao** [comment]
List wished features so that people can have more overview about what is still missing and if we plan it.

**Reply by haixuanTao** [reply]
Added wished feature in https://github.com/dora-rs/dora/pull/316/commits/3b0f69f07de8ce434090a9ed1627dc030e99c3ee

**Comment by haixuanTao** [comment]
Depreciate existing documentation and README and point to the new documentation

**Reply by haixuanTao** [reply]
Opened #316

**Comment by haixuanTao** [comment]
Add video demo of installation that can later be used for demos
